Project-21-React-Animation/src/components/Challenges.jsx

```diff
@@ -0,0 +1,80 @@
+import { useContext, useState } from "react";
+import { AnimatePresence, motion } from "framer-motion";
+
+import { ChallengesContext } from "../store/challenges-context.jsx";
+import ChallengeItem from "./ChallengeItem.jsx";
+import ChallengeTabs from "./ChallengeTabs.jsx";
+
+export default function Challenges() {
+  const { challenges } = useContext(ChallengesContext);
+  const [selectedType, setSelectedType] = useState("active");
+  const [expanded, setExpanded] = useState(null);
+
+  function handleSelectType(newType) {
+    setSelectedType(newType);
+  }
+
+  function handleViewDetails(id) {
+    setExpanded((prevId) => {
+      if (prevId === id) {
+        return null;
+      }
+
+      return id;
+    });
+  }
+
+  const filteredChallenges = {
+    active: challenges.filter((challenge) => challenge.status === "active"),
+    completed: challenges.filter(
+      (challenge) => challenge.status === "completed"
+    ),
+    failed: challenges.filter((challenge) => challenge.status === "failed"),
+  };
```

Comment:
@sknepali The filteredChallenges object is recalculated on every render, even if the challenges array hasn't changed. This could be optimized using the useMemo hook to only recalculate when challenges changes.